artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -570,8 +569,13 @@ private void connectSender(Queue queue,
                      Object remoteBrokerID = sender.getRemoteProperties().get(AMQPMirrorControllerSource.BROKER_ID);
                      if (remoteBrokerID.equals(brokerID)) {
                         error(ActiveMQAMQPProtocolMessageBundle.BUNDLE.brokerConnectionMirrorItself(), lastRetryCounter);
+                        return;
                      }
                   }
+                  if (sender.getRemoteTarget() == null) {
+                     error(ActiveMQAMQPProtocolMessageBundle.BUNDLE.missingRemoteTarget(sender.getTarget().getAddress()), lastRetryCounter);
```

Comment:
error here will cause a disconnect. Nothing else should arrive after that. I can probably clean it up just in case, I could do it for everything on error.

Reply:
It will, but does nothing about thigns that already arrived. Its likely the following detach already arrived before the attach is even processed, and so can/will be processed immediately following this.